core/src/main/java/software/amazon/awssdk/http/pipeline/stages/ApplyUserAgentStage.java

```diff
@@ -31,20 +32,38 @@
 public class ApplyUserAgentStage implements MutableRequestToRequestPipeline {
 
     private final LegacyClientConfiguration config;
+    private final String clientName;
+    private final String clientType;
 
-    public ApplyUserAgentStage(HttpClientDependencies dependencies) {
+    public ApplyUserAgentStage(HttpClientDependencies dependencies,
+                               String clientName,
+                               String clientType) {
```

Comment:
The type is not obvious if we support Apache Async.

You can't easily get the client name from the dependencies as ApplyUserAgentStage doesn't know if the sync or async client is being used though presumably only 1 will be non-null.